app/scripts/components/exploration/atoms.ts

```diff
@@ -0,0 +1,37 @@
+import { atom } from 'jotai';
+import {
+  DateRange,
+  HEADER_COLUMN_WIDTH,
+  RIGHT_AXIS_SPACE,
+  TimelineDataset,
+  ZoomTransformPlain
+} from './constants';
+
+// Datasets to show on the timeline and their settings
+export const timelineDatasetsAtom = atom<TimelineDataset[]>([]);
+// Main timeline date. This date defines the datasets shown on the map.
+export const selectedDateAtom = atom<Date | null>(null);
+// Date range for L&R playheads.
+export const selectedIntervalAtom = atom<DateRange | null>(null);
+// Zoom transform for the timeline. Values as object instead of d3.ZoomTransform
+export const zoomTransformAtom = atom<ZoomTransformPlain>({
+  x: 0,
+  y: 0,
+  k: 1
+});
+// Width of the whole timeline item. Set via a size observer and then used to
+// compute the different element sizes.
+export const timelineWidthAtom = atom<number | undefined>(undefined);
```

Comment:
I think I could use a high-level explanation as to why this is part of the app state. Isn't that defined solely by the available witdh of the timeline viewport?

Reply:
There are several elements that need to know the width (specially the svg components), so I thought of storing the size in an atom and access it via a hook instead of having to pass width to every single component (this is something I still have to refactor) - Thoughts on the approach?